lib/devise/doorkeeper/doorkeeper_failure_app.rb

```diff
@@ -3,9 +3,25 @@
 module Devise
   module Doorkeeper
     module DoorkeeperFailureApp
+      class UnconfirmedResourceResponse < ::Doorkeeper::OAuth::ErrorResponse
```

Comment:
Minor: Would it make sense to move this into a new file like `lib/doorkeeper/oauth/unconfirmed_resource_response.rb` ?

Reply:
I think that's better given how they're structured inside doorkeeper